packages/trader/src/AppV2/Components/ServiceErrorSheet/service-error-sheet.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from 'react';
+import { observer, useStore } from '@deriv/stores';
+import { ActionSheet, Text } from '@deriv-com/quill-ui';
+import { getLanguage, Localize } from '@deriv/translations';
+import { redirectToLogin } from '@deriv/shared';
+import { useSignupHandler } from 'AppV2/Hooks/useSignUpHandler';
+import './service-error-sheet.scss';
+import { useHistory } from 'react-router';
+
+const ServiceErrorSheet = observer(() => {
+    const [is_open, setIsOpen] = useState(false);
+    const { common } = useStore();
+    const { services_error, resetServicesError } = common;
+    const { handleSignup } = useSignupHandler();
+    const history = useHistory();
+
+    const is_insufficient_balance =
+        services_error.code === 'InsufficientBalance' || services_error.code === 'InvalidContractProposal';
+    const is_authorization_required = services_error.code === 'AuthorizationRequired' && services_error.type === 'buy';
+
+    useEffect(() => {
+        if (is_insufficient_balance || is_authorization_required) {
+            setIsOpen(true);
+        }
+    }, [services_error]);
+
+    useEffect(() => {
+        if (!is_open && services_error.code) {
+            resetServicesError();
+        }
+    }, [resetServicesError, is_open]);
```

Comment:
Do we need `resetServicesError` in array of dep-s?

Reply:
It doesn't make any difference .  My eslint extension keeps on pinging me to add all methods that are used inside useEffect.